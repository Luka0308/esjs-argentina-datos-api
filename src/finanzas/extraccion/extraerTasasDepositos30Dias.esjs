importar { load } desde 'cheerio'
importar { format, parse } desde 'date-fns'
importar { extraerBcra } desde '@/extractores/bcra.extractor.esjs'

exportar asincrono funcion extraerTasasDepositos30Dias(desde, hasta) {
  const html = esperar extraerBcra('1212', desde, hasta)

  const $ = load(html)

  const tabla = $('table')

  const items = []

  tabla.buscar('tbody tr').each((i, tr) => {
    const fecha = $(tr).buscar('td').eq(0).text()
    const valor = $(tr).buscar('td').eq(1).text()

    items.agregar({
      fecha: format(parse(fecha, 'dd/MM/yyyy', crear Fecha()), 'yyyy-MM-dd'),
      valor: interpretarPorcentaje(valor),
    })
  })

  retornar items
}

funcion interpretarPorcentaje(valor) {
  const porcentaje = Numero(valor.reemplazar(',', '.').reemplazar('%', ''))

  si (!porcentaje) {
    retornar nulo
  }

  retornar interpretarDecimal((Numero(porcentaje) / 100).fijarDecimales(4));
}
