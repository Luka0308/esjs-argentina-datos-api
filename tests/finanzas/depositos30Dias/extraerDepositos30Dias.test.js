import { describe, expect, it } from 'vitest'
import { extraerTasasDepositos30Dias } from '@/finanzas/extraccion/extraerTasasDepositos30Dias.esjs'
import { format } from 'date-fns'

describe('extraerTasasDepositos30Dias', () => {
  it('extrae los depósitos a 30 días', async () => {
    const items = await extraerTasasDepositos30Dias(
      '2023-01-01',
      '2024-01-01'
    )

    expect(items).toMatchSnapshot()
  })
})
