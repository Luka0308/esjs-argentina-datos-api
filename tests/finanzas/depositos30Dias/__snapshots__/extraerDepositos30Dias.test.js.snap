// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`extraerTasasDepositos30Dias > extrae los depósitos a 30 días 1`] = `
[
  {
    "fecha": "2023-01-02",
    "valor": 0.6867,
  },
  {
    "fecha": "2023-01-03",
    "valor": 0.7046,
  },
  {
    "fecha": "2023-01-04",
    "valor": 0.7083,
  },
  {
    "fecha": "2023-01-05",
    "valor": 0.7025,
  },
  {
    "fecha": "2023-01-06",
    "valor": 0.7078,
  },
  {
    "fecha": "2023-01-09",
    "valor": 0.7046,
  },
  {
    "fecha": "2023-01-10",
    "valor": 0.7006,
  },
  {
    "fecha": "2023-01-11",
    "valor": 0.7032,
  },
  {
    "fecha": "2023-01-12",
    "valor": 0.7005,
  },
  {
    "fecha": "2023-01-13",
    "valor": 0.6946,
  },
  {
    "fecha": "2023-01-16",
    "valor": 0.7109,
  },
  {
    "fecha": "2023-01-17",
    "valor": 0.7052,
  },
  {
    "fecha": "2023-01-18",
    "valor": 0.7038,
  },
  {
    "fecha": "2023-01-19",
    "valor": 0.7057,
  },
  {
    "fecha": "2023-01-20",
    "valor": 0.707,
  },
  {
    "fecha": "2023-01-23",
    "valor": 0.7098,
  },
  {
    "fecha": "2023-01-24",
    "valor": 0.7102,
  },
  {
    "fecha": "2023-01-25",
    "valor": 0.6978,
  },
  {
    "fecha": "2023-01-26",
    "valor": 0.7083,
  },
  {
    "fecha": "2023-01-27",
    "valor": 0.6955,
  },
  {
    "fecha": "2023-01-30",
    "valor": 0.7083,
  },
  {
    "fecha": "2023-01-31",
    "valor": 0.7087,
  },
  {
    "fecha": "2023-02-01",
    "valor": 0.7013,
  },
  {
    "fecha": "2023-02-02",
    "valor": 0.6785,
  },
  {
    "fecha": "2023-02-03",
    "valor": 0.7006,
  },
  {
    "fecha": "2023-02-06",
    "valor": 0.7085,
  },
  {
    "fecha": "2023-02-07",
    "valor": 0.6936,
  },
  {
    "fecha": "2023-02-08",
    "valor": 0.7117,
  },
  {
    "fecha": "2023-02-09",
    "valor": 0.7111,
  },
  {
    "fecha": "2023-02-10",
    "valor": 0.6958,
  },
  {
    "fecha": "2023-02-13",
    "valor": 0.7126,
  },
  {
    "fecha": "2023-02-14",
    "valor": 0.6868,
  },
  {
    "fecha": "2023-02-15",
    "valor": 0.7078,
  },
  {
    "fecha": "2023-02-16",
    "valor": 0.7041,
  },
  {
    "fecha": "2023-02-17",
    "valor": 0.7051,
  },
  {
    "fecha": "2023-02-22",
    "valor": 0.7108,
  },
  {
    "fecha": "2023-02-23",
    "valor": 0.7167,
  },
  {
    "fecha": "2023-02-24",
    "valor": 0.7008,
  },
  {
    "fecha": "2023-02-27",
    "valor": 0.7062,
  },
  {
    "fecha": "2023-02-28",
    "valor": 0.7014,
  },
  {
    "fecha": "2023-03-01",
    "valor": 0.707,
  },
  {
    "fecha": "2023-03-02",
    "valor": 0.7061,
  },
  {
    "fecha": "2023-03-03",
    "valor": 0.7028,
  },
  {
    "fecha": "2023-03-06",
    "valor": 0.7063,
  },
  {
    "fecha": "2023-03-07",
    "valor": 0.6915,
  },
  {
    "fecha": "2023-03-08",
    "valor": 0.71,
  },
  {
    "fecha": "2023-03-09",
    "valor": 0.7133,
  },
  {
    "fecha": "2023-03-10",
    "valor": 0.6988,
  },
  {
    "fecha": "2023-03-13",
    "valor": 0.7063,
  },
  {
    "fecha": "2023-03-14",
    "valor": 0.6831,
  },
  {
    "fecha": "2023-03-15",
    "valor": 0.703,
  },
  {
    "fecha": "2023-03-16",
    "valor": 0.7079,
  },
  {
    "fecha": "2023-03-17",
    "valor": 0.7287,
  },
  {
    "fecha": "2023-03-20",
    "valor": 0.7316,
  },
  {
    "fecha": "2023-03-21",
    "valor": 0.7163,
  },
  {
    "fecha": "2023-03-22",
    "valor": 0.7267,
  },
  {
    "fecha": "2023-03-23",
    "valor": 0.7224,
  },
  {
    "fecha": "2023-03-27",
    "valor": 0.7353,
  },
  {
    "fecha": "2023-03-28",
    "valor": 0.7301,
  },
  {
    "fecha": "2023-03-29",
    "valor": 0.7272,
  },
  {
    "fecha": "2023-03-30",
    "valor": 0.7282,
  },
  {
    "fecha": "2023-03-31",
    "valor": 0.7358,
  },
  {
    "fecha": "2023-04-03",
    "valor": 0.736,
  },
  {
    "fecha": "2023-04-04",
    "valor": 0.7286,
  },
  {
    "fecha": "2023-04-05",
    "valor": 0.7316,
  },
  {
    "fecha": "2023-04-10",
    "valor": 0.7377,
  },
  {
    "fecha": "2023-04-11",
    "valor": 0.733,
  },
  {
    "fecha": "2023-04-12",
    "valor": 0.7291,
  },
  {
    "fecha": "2023-04-13",
    "valor": 0.7371,
  },
  {
    "fecha": "2023-04-14",
    "valor": 0.7302,
  },
  {
    "fecha": "2023-04-17",
    "valor": 0.7396,
  },
  {
    "fecha": "2023-04-18",
    "valor": 0.7352,
  },
  {
    "fecha": "2023-04-19",
    "valor": 0.736,
  },
  {
    "fecha": "2023-04-20",
    "valor": 0.7345,
  },
  {
    "fecha": "2023-04-21",
    "valor": 0.7469,
  },
  {
    "fecha": "2023-04-24",
    "valor": 0.7564,
  },
  {
    "fecha": "2023-04-25",
    "valor": 0.748,
  },
  {
    "fecha": "2023-04-26",
    "valor": 0.7613,
  },
  {
    "fecha": "2023-04-27",
    "valor": 0.7634,
  },
  {
    "fecha": "2023-04-28",
    "valor": 0.8757,
  },
  {
    "fecha": "2023-05-02",
    "valor": 0.8762,
  },
  {
    "fecha": "2023-05-03",
    "valor": 0.8755,
  },
  {
    "fecha": "2023-05-04",
    "valor": 0.8568,
  },
  {
    "fecha": "2023-05-05",
    "valor": 0.8769,
  },
  {
    "fecha": "2023-05-08",
    "valor": 0.8598,
  },
  {
    "fecha": "2023-05-09",
    "valor": 0.8639,
  },
  {
    "fecha": "2023-05-10",
    "valor": 0.8779,
  },
  {
    "fecha": "2023-05-11",
    "valor": 0.8826,
  },
  {
    "fecha": "2023-05-12",
    "valor": 0.8747,
  },
  {
    "fecha": "2023-05-15",
    "valor": 0.8862,
  },
  {
    "fecha": "2023-05-16",
    "valor": 0.9264,
  },
  {
    "fecha": "2023-05-17",
    "valor": 0.9325,
  },
  {
    "fecha": "2023-05-18",
    "valor": 0.9257,
  },
  {
    "fecha": "2023-05-19",
    "valor": 0.928,
  },
  {
    "fecha": "2023-05-22",
    "valor": 0.9246,
  },
  {
    "fecha": "2023-05-23",
    "valor": 0.9156,
  },
  {
    "fecha": "2023-05-24",
    "valor": 0.9111,
  },
  {
    "fecha": "2023-05-29",
    "valor": 0.9307,
  },
  {
    "fecha": "2023-05-30",
    "valor": 0.9208,
  },
  {
    "fecha": "2023-05-31",
    "valor": 0.9083,
  },
  {
    "fecha": "2023-06-01",
    "valor": 0.9255,
  },
  {
    "fecha": "2023-06-02",
    "valor": 0.9323,
  },
  {
    "fecha": "2023-06-05",
    "valor": 0.9263,
  },
  {
    "fecha": "2023-06-06",
    "valor": 0.9292,
  },
  {
    "fecha": "2023-06-07",
    "valor": 0.9113,
  },
  {
    "fecha": "2023-06-08",
    "valor": 0.9299,
  },
  {
    "fecha": "2023-06-09",
    "valor": 0.9235,
  },
  {
    "fecha": "2023-06-12",
    "valor": 0.9356,
  },
  {
    "fecha": "2023-06-13",
    "valor": 0.928,
  },
  {
    "fecha": "2023-06-14",
    "valor": 0.9327,
  },
  {
    "fecha": "2023-06-15",
    "valor": 0.9321,
  },
  {
    "fecha": "2023-06-16",
    "valor": 0.9268,
  },
  {
    "fecha": "2023-06-21",
    "valor": 0.9351,
  },
  {
    "fecha": "2023-06-22",
    "valor": 0.9225,
  },
  {
    "fecha": "2023-06-23",
    "valor": 0.9147,
  },
  {
    "fecha": "2023-06-26",
    "valor": 0.9379,
  },
  {
    "fecha": "2023-06-27",
    "valor": 0.9188,
  },
  {
    "fecha": "2023-06-28",
    "valor": 0.9288,
  },
  {
    "fecha": "2023-06-29",
    "valor": 0.9212,
  },
  {
    "fecha": "2023-06-30",
    "valor": 0.9305,
  },
  {
    "fecha": "2023-07-03",
    "valor": 0.9328,
  },
  {
    "fecha": "2023-07-04",
    "valor": 0.9291,
  },
  {
    "fecha": "2023-07-05",
    "valor": 0.9307,
  },
  {
    "fecha": "2023-07-06",
    "valor": 0.9355,
  },
  {
    "fecha": "2023-07-07",
    "valor": 0.9245,
  },
  {
    "fecha": "2023-07-10",
    "valor": 0.9437,
  },
  {
    "fecha": "2023-07-11",
    "valor": 0.9268,
  },
  {
    "fecha": "2023-07-12",
    "valor": 0.9345,
  },
  {
    "fecha": "2023-07-13",
    "valor": 0.9346,
  },
  {
    "fecha": "2023-07-14",
    "valor": 0.9304,
  },
  {
    "fecha": "2023-07-17",
    "valor": 0.9368,
  },
  {
    "fecha": "2023-07-18",
    "valor": 0.9301,
  },
  {
    "fecha": "2023-07-19",
    "valor": 0.9262,
  },
  {
    "fecha": "2023-07-20",
    "valor": 0.9155,
  },
  {
    "fecha": "2023-07-21",
    "valor": 0.9327,
  },
  {
    "fecha": "2023-07-24",
    "valor": 0.9341,
  },
  {
    "fecha": "2023-07-25",
    "valor": 0.9257,
  },
  {
    "fecha": "2023-07-26",
    "valor": 0.9261,
  },
  {
    "fecha": "2023-07-27",
    "valor": 0.9173,
  },
  {
    "fecha": "2023-07-28",
    "valor": 0.9279,
  },
  {
    "fecha": "2023-07-31",
    "valor": 0.9359,
  },
  {
    "fecha": "2023-08-01",
    "valor": 0.9166,
  },
  {
    "fecha": "2023-08-02",
    "valor": 0.9258,
  },
  {
    "fecha": "2023-08-03",
    "valor": 0.9296,
  },
  {
    "fecha": "2023-08-04",
    "valor": 0.9347,
  },
  {
    "fecha": "2023-08-07",
    "valor": 0.9381,
  },
  {
    "fecha": "2023-08-08",
    "valor": 0.9238,
  },
  {
    "fecha": "2023-08-09",
    "valor": 0.9373,
  },
  {
    "fecha": "2023-08-10",
    "valor": 0.9359,
  },
  {
    "fecha": "2023-08-11",
    "valor": 0.9329,
  },
  {
    "fecha": "2023-08-14",
    "valor": 0.953,
  },
  {
    "fecha": "2023-08-15",
    "valor": 1.136,
  },
  {
    "fecha": "2023-08-16",
    "valor": 1.1382,
  },
  {
    "fecha": "2023-08-17",
    "valor": 1.1295,
  },
  {
    "fecha": "2023-08-18",
    "valor": 1.1244,
  },
  {
    "fecha": "2023-08-22",
    "valor": 1.1253,
  },
  {
    "fecha": "2023-08-23",
    "valor": 1.1354,
  },
  {
    "fecha": "2023-08-24",
    "valor": 1.1307,
  },
  {
    "fecha": "2023-08-25",
    "valor": 1.1338,
  },
  {
    "fecha": "2023-08-28",
    "valor": 1.1387,
  },
  {
    "fecha": "2023-08-29",
    "valor": 1.124,
  },
  {
    "fecha": "2023-08-30",
    "valor": 1.1248,
  },
  {
    "fecha": "2023-08-31",
    "valor": 1.1363,
  },
  {
    "fecha": "2023-09-01",
    "valor": 1.1366,
  },
  {
    "fecha": "2023-09-04",
    "valor": 1.1271,
  },
  {
    "fecha": "2023-09-05",
    "valor": 1.1242,
  },
  {
    "fecha": "2023-09-06",
    "valor": 1.1425,
  },
  {
    "fecha": "2023-09-07",
    "valor": 1.1434,
  },
  {
    "fecha": "2023-09-08",
    "valor": 1.1495,
  },
  {
    "fecha": "2023-09-11",
    "valor": 1.1349,
  },
  {
    "fecha": "2023-09-12",
    "valor": 1.1395,
  },
  {
    "fecha": "2023-09-13",
    "valor": 1.1298,
  },
  {
    "fecha": "2023-09-14",
    "valor": 1.1351,
  },
  {
    "fecha": "2023-09-15",
    "valor": 1.1404,
  },
  {
    "fecha": "2023-09-18",
    "valor": 1.1326,
  },
  {
    "fecha": "2023-09-19",
    "valor": 1.1328,
  },
  {
    "fecha": "2023-09-20",
    "valor": 1.1073,
  },
  {
    "fecha": "2023-09-21",
    "valor": 1.1029,
  },
  {
    "fecha": "2023-09-22",
    "valor": 1.1389,
  },
  {
    "fecha": "2023-09-25",
    "valor": 1.1403,
  },
  {
    "fecha": "2023-09-26",
    "valor": 1.1289,
  },
  {
    "fecha": "2023-09-27",
    "valor": 1.134,
  },
  {
    "fecha": "2023-09-28",
    "valor": 1.137,
  },
  {
    "fecha": "2023-09-29",
    "valor": 1.1193,
  },
  {
    "fecha": "2023-10-02",
    "valor": 1.1404,
  },
  {
    "fecha": "2023-10-03",
    "valor": 1.1182,
  },
  {
    "fecha": "2023-10-04",
    "valor": 1.1165,
  },
  {
    "fecha": "2023-10-05",
    "valor": 1.1414,
  },
  {
    "fecha": "2023-10-06",
    "valor": 1.1352,
  },
  {
    "fecha": "2023-10-09",
    "valor": 1.1512,
  },
  {
    "fecha": "2023-10-10",
    "valor": 1.1429,
  },
  {
    "fecha": "2023-10-11",
    "valor": 1.1386,
  },
  {
    "fecha": "2023-10-12",
    "valor": 1.1454,
  },
  {
    "fecha": "2023-10-17",
    "valor": 1.2917,
  },
  {
    "fecha": "2023-10-18",
    "valor": 1.2892,
  },
  {
    "fecha": "2023-10-19",
    "valor": 1.279,
  },
  {
    "fecha": "2023-10-20",
    "valor": 1.2868,
  },
  {
    "fecha": "2023-10-23",
    "valor": 1.2874,
  },
  {
    "fecha": "2023-10-24",
    "valor": 1.273,
  },
  {
    "fecha": "2023-10-25",
    "valor": 1.2855,
  },
  {
    "fecha": "2023-10-26",
    "valor": 1.2782,
  },
  {
    "fecha": "2023-10-27",
    "valor": 1.2819,
  },
  {
    "fecha": "2023-10-30",
    "valor": 1.2926,
  },
  {
    "fecha": "2023-10-31",
    "valor": 1.2761,
  },
  {
    "fecha": "2023-11-01",
    "valor": 1.2895,
  },
  {
    "fecha": "2023-11-02",
    "valor": 1.2819,
  },
  {
    "fecha": "2023-11-03",
    "valor": 1.2965,
  },
  {
    "fecha": "2023-11-07",
    "valor": 1.3007,
  },
  {
    "fecha": "2023-11-08",
    "valor": 1.2951,
  },
  {
    "fecha": "2023-11-09",
    "valor": 1.2965,
  },
  {
    "fecha": "2023-11-10",
    "valor": 1.2936,
  },
  {
    "fecha": "2023-11-13",
    "valor": 1.3042,
  },
  {
    "fecha": "2023-11-14",
    "valor": 1.289,
  },
  {
    "fecha": "2023-11-15",
    "valor": 1.2795,
  },
  {
    "fecha": "2023-11-16",
    "valor": 1.2915,
  },
  {
    "fecha": "2023-11-17",
    "valor": 1.2929,
  },
  {
    "fecha": "2023-11-21",
    "valor": 1.2989,
  },
  {
    "fecha": "2023-11-22",
    "valor": 1.2891,
  },
  {
    "fecha": "2023-11-23",
    "valor": 1.2712,
  },
  {
    "fecha": "2023-11-24",
    "valor": 1.2918,
  },
  {
    "fecha": "2023-11-27",
    "valor": 1.2885,
  },
  {
    "fecha": "2023-11-28",
    "valor": 1.279,
  },
  {
    "fecha": "2023-11-29",
    "valor": 1.2892,
  },
  {
    "fecha": "2023-11-30",
    "valor": 1.2794,
  },
  {
    "fecha": "2023-12-01",
    "valor": 1.2892,
  },
  {
    "fecha": "2023-12-04",
    "valor": 1.2934,
  },
  {
    "fecha": "2023-12-05",
    "valor": 1.2884,
  },
  {
    "fecha": "2023-12-06",
    "valor": 1.2843,
  },
  {
    "fecha": "2023-12-07",
    "valor": 1.2974,
  },
  {
    "fecha": "2023-12-11",
    "valor": 1.2957,
  },
  {
    "fecha": "2023-12-12",
    "valor": 1.2909,
  },
  {
    "fecha": "2023-12-13",
    "valor": 1.2927,
  },
  {
    "fecha": "2023-12-14",
    "valor": 1.1987,
  },
  {
    "fecha": "2023-12-15",
    "valor": 1.2868,
  },
  {
    "fecha": "2023-12-18",
    "valor": 1.2966,
  },
  {
    "fecha": "2023-12-19",
    "valor": 1.1584,
  },
  {
    "fecha": "2023-12-20",
    "valor": 1.094,
  },
  {
    "fecha": "2023-12-21",
    "valor": 1.0953,
  },
  {
    "fecha": "2023-12-22",
    "valor": 1.0894,
  },
  {
    "fecha": "2023-12-26",
    "valor": 1.0777,
  },
  {
    "fecha": "2023-12-27",
    "valor": 1.0803,
  },
  {
    "fecha": "2023-12-28",
    "valor": 1.0867,
  },
  {
    "fecha": "2023-12-29",
    "valor": 1.0699,
  },
]
`;
